operators/azure/cloud_base_kernel.hpp

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "ocos.h"
+#include "gsl/span"
+
+namespace ort_extensions {
+
+/// <summary>
+/// Base kernel for custom ops that call cloud endpoints.
+/// </summary>
+class CloudBaseKernel : public BaseKernel {
+ protected:
+  CloudBaseKernel(const OrtApi& api, const OrtKernelInfo& info);
+  virtual ~CloudBaseKernel() = default;
+
+  // Names of attributes the custom operator provides.
+  static constexpr const char* kUri = "model_uri";           // required
+  static constexpr const char* kModelName = "model_name";    // optional
+  static constexpr const char* kModelVer = "model_version";  // optional
+  static constexpr const char* kVerbose = "verbose";
+
+  static constexpr int MinimumSupportedOrtVersion = 14;
+
+  const std::string& ModelUri() const { return model_uri_; }
+  const std::string& ModelName() const { return model_name_; }
+  const std::string& ModelVersion() const { return model_ver_; }
+  bool Verbose() const { return verbose_; }
+
+  const gsl::span<const std::string> InputNames() const { return input_names_; }
+  const gsl::span<const std::string> OutputNames() const { return output_names_; }
+
+  // Request property names that are parsed from input names. 1:1 with InputNames() values.
+  const gsl::span<const std::string> PropertyNames() const { return property_names_; }
```

Comment:
It's not quite clear what a property is. I know you mean request paramters sent in the HTTPS request. Might be clearer to spell it out.